zx/compare.go

```diff
@@ -143,6 +151,46 @@ func CompareInt64(op string, pattern int64) (Predicate, error) {
 	}, nil
 }
 
+func CompareUint64(op string, pattern uint64) (Predicate, error) {
+	CompareUint, ok1 := compareUint[op]
+	CompareFloat, ok2 := compareFloat[op]
+	if !ok1 || !ok2 {
+		return nil, fmt.Errorf("unknown int comparator: %s", op)
+	}
+	// many different zeek data types can be compared with integers
+	return func(val zng.Value) bool {
+		zv := val.Bytes
+		switch val.Type.(type) {
+		case *zng.TypeOfInt16, *zng.TypeOfInt32, *zng.TypeOfInt64:
```

Comment:
You should also switch on `val.Type.ID()` here as well to avoid the same alias issue.

Reply:
Given that comparisons aren't currently working in master, do you mind if I make that change in #326?  (which I still need to update to use `ID()` instead of `.type()`)

Reply:
Sounds good!